frontend/src/components/Navbar/Navbar.tsx

```diff
@@ -1,21 +1,37 @@
+import { createPortal } from "react-dom";
 import classes from "./Navbar.module.scss";
 import Logo from "../Logo/Logo";
 import ProfileIcon from "../ProfileIcon/ProfileIcon";
+import { useState } from "react";
+import Credits from "../Credits/Credits";
 
 function Navbar() {
+  const [showCredits, setShowCredits] = useState(false);
+
+  const toggleCredits = () => {
+    setShowCredits((prev) => !prev);
+  };
+
   return (
     <nav className={classes.navbar}>
       <div>
         <button className={`${classes.hover} ${classes.logo}`}>
-          <Logo size="lg"/>
+          <Logo size="lg" />
         </button>
         <button className={classes.hover}>Gamemodes</button>
         <button className={classes.hover}>Help</button>
       </div>
       <div>
-        <button className={classes.hover}>Credits</button>
+        <button className={classes.hover} onClick={toggleCredits}>
+          Credits
+        </button>
         <ProfileIcon url="/yellowshirt.svg" />
       </div>
+      {showCredits &&
```

Comment:
i do worry this approach is a bit less readable than just using a native <dialog> but its all good